### PR TITLE
fix(op): filter out receipts for dup txns

### PR DESCRIPTION
**Pull request by emhane**
- Filters out receipts for duplicate transactions, which have been filtered out for txns already
- Moves import bedrock receipts to new command
- Fixes tracing
- Coherency checks, consistency with transaction static files

After importing receipts and blocks + init state
Screenshot 2024-05-25 at 16 18 48

After syncing 1M blocks
Screenshot 2024-05-25 at 19 18 20

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -43,6 +43,7 @@ members = [
     "crates/optimism/evm/",
     "crates/optimism/node/",
     "crates/optimism/payload/",
+    "crates/optimism/primitives/",
```
Comment (emhane) [inline Cargo.toml:46]: let me know if I should remove this and put the type contained in the crate back in `reth-primitives`, wanted to make a move in this direction @mattsse

File: bin/reth/src/commands/import_receipts_op.rs
Comment (mattsse) [inline bin/reth/src/commands/import_receipts_op.rs:1]: why is this a new file now?

File: crates/optimism/primitives/Cargo.toml
Comment (mattsse) [inline crates/optimism/primitives/Cargo.toml:1]: this makes sense in general, but is unrelated to this fix?

**Review by mattsse — COMMENTED** [review]
not obvious what is fix and what is refactor here.

the import receipt cmd is only useful for op mainnet, I don't think maintaining two versions of this command is useful

File: crates/optimism/primitives/Cargo.toml
Comment (emhane) [reply crates/optimism/primitives/Cargo.toml:1]: kind of unrelated

**Comment by emhane** [comment]
> not obvious what is fix and what is refactor here.
> 
> the import receipt cmd is only useful for op mainnet, I don't think maintaining two versions of this command is useful

op is the only use case we have though of at least, happy if reth is used in other ways for example by researchers. in testing, the fact that `import`, `import-receipts` and `init-state` are generic commands, allows starting the chain at any given height

File: bin/reth/src/commands/import_receipts_op.rs
Comment (emhane) [reply bin/reth/src/commands/import_receipts_op.rs:1]: hm realised this issue must be addressed to make it fully generic https://github.com/paradigmxyz/reth/issues/8145, i.e. allow import of receipts exported in standard rlp format and not on the branch of testinprod

**Comment by emhane** [comment]
scope of the pr grew a bit to update the book @mattsse

File: book/run/sync-op-mainnet.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ Output from running the command to export state, can also be downloaded from <ht
 
 Imports a `.rlp` file of blocks.
 
-Note! Requires running in debug mode (TODO: <https://github.com/paradigmxyz/reth/issues/7650>).
+Import of >100 million OVM blocks, from genesis to Bedrock, completes in 6 hours.
```
Comment (emhane) [inline book/run/sync-op-mainnet.md:25]: running in maxperf now, let me see if this takes significantly less time first

File: crates/optimism/primitives/src/lib.rs
Comment (mattsse) [inline crates/optimism/primitives/src/lib.rs:1]: still needs the regular header

**Review by mattsse — CHANGES_REQUESTED** [review]
pedantic style nit

**Review by mattsse — APPROVED** [review]

**Pull request merged**